lib/pure/strutils.nim

```diff
@@ -84,31 +84,41 @@ from std/private/strimpl import cmpIgnoreStyleImpl, cmpIgnoreCaseImpl, startsWit
 
 
 const
-  Whitespace* = {' ', '\t', '\v', '\r', '\l', '\f'}
+  Whitespace* = {' ', '\t', '\v', '\r', '\n', '\f'}
     ## All the characters that count as whitespace (space, tab, vertical tab,
     ## carriage return, new line, form feed).
 
   Letters* = {'A'..'Z', 'a'..'z'}
     ## The set of letters.
 
+  Vowels* = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
+    ## The set of vowels.
+  
+  Consonants* = Letters - Vowels
+    ## The set of consonants.
+
   Digits* = {'0'..'9'}
     ## The set of digits.
 
-  HexDigits* = {'0'..'9', 'A'..'F', 'a'..'f'}
+  HexDigits* = Digits + {'A'..'F', 'a'..'f'}
     ## The set of hexadecimal digits.
 
-  IdentChars* = {'a'..'z', 'A'..'Z', '0'..'9', '_'}
+  IdentChars* = Letters + Digits + {'_'}
     ## The set of characters an identifier can consist of.
 
-  IdentStartChars* = {'a'..'z', 'A'..'Z', '_'}
+  IdentStartChars* = Letters + {'_'}
     ## The set of characters an identifier can start with.
 
-  Newlines* = {'\13', '\10'}
-    ## The set of characters a newline terminator can start with (carriage
-    ## return, line feed).
+  Newlines* = {'\r', '\n'}
+    ## The set of characters a newline terminator can consist of
+
+  # same as {'!', '\"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',', '-', '.', '/', ':', ';', '<', '=', '>', '?', '@', '[', '\\', ']', '^', '_', '`', '{', '|', '}', '~'}
+  # same as {'!'..'/', ':'..'@', '['..'`', '{'..'~'}
+  Punctuation* = {'!'..'~'} - Letters - Digits
```

Comment:
why not give the list so it's self-documenting?

```
  Punctuation* = {'!', '\"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',', '-', '.', '/', ':', ';', '<', '=', '>', '?', '@', '[', '\\', ']', '^', '_', '`', '{', '|', '}', '~'}
```

Reply:
Yeah, I know that the performance is the same... Since, as far as I know, the set differences would be computed compile time, so those two are literally the same.
Gonna do another commit, and change it to that, but I was just unsure which one makes it more obvious. Since the version I left there shows that that means all characters except alphanumeric.
Changing it to `GraphicChars - Letters - Digits` might make more sense, when it's added